src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicConfig.cs

```diff
@@ -237,33 +172,15 @@ public override void ExecuteSiteRecoveryCmdlet()
             base.ExecuteSiteRecoveryCmdlet();
             ASRVMNicConfig nicConfig = null;
 
-            if (string.IsNullOrEmpty(this.RecoveryVMNetworkId) &&
-                !string.IsNullOrEmpty(this.RecoveryVMSubnetName))
-            {
-                this.WriteWarning(Resources.RecoveryNetworkInformationMissing);
-                return;
-            }
-
-            if (string.IsNullOrEmpty(this.TfoVMNetworkId) &&
-                !string.IsNullOrEmpty(this.TfoVMSubnetName))
-            {
-                this.WriteWarning(Resources.TfoNetworkInformationMissing);
-                return;
-            }
-
-            if (string.IsNullOrEmpty(this.RecoveryVMSubnetName) &&
-                !string.IsNullOrEmpty(this.RecoveryNicStaticIPAddress))
-            {
-                this.WriteWarning(Resources.RecoverySubnetInformationMissing);
-                return;
-            }
 
-            if (string.IsNullOrEmpty(this.TfoVMSubnetName) &&
-                !string.IsNullOrEmpty(this.TfoNicStaticIPAddress))
+            // how to verify if tfo/fo update input is provided and check corresponding network id.
```

Comment:
Check the condition, comment and the associated warning. The warning says recovery info missing but is being printed in case of a non-empty recovery network id.

Reply:
existing logic had access to recovery/tfo subnet name.. hence can infer whether recovery/tfo network id is required and raise error if they are not provided.. but now  i moved them to ipconfigs..  so even if tfonetwork is not provided (if needed) given we are populating defaults from RPI in the NewAzureRmAsrVmNicIPConfig cmdlet, there is no info to check whether cx provided tfoNetworkId and its from the RPI.. i need to introduce a flag to track that from NewAzureRmAsrVmNicIPConfig  cmdlet (same for recovery network id). Correct me if i am missing something.

Reply:
In that case, we need to add this validation to the IP config cmdlet and remove it from here

Reply:
moved validation and defaulting logic to NicConfig cmdlet as discussed to accommodate network validation.